lib/core/player.rb

```diff
@@ -0,0 +1,22 @@
+require 'UI/UI'
+require 'core/board'
+
+class Player
+  attr_accessor :symbol
+  attr_reader :ui
+
+  def initialize(symbol, ui)
+    @symbol = symbol
+    @ui = ui
+  end
+
+  def get_valid_position(board)
+    position = @ui.get_position
+    until board.grid[position].is_empty? do
```

Comment:
Maybe extract out into error function?

Reply:
Like this: ```
until board.grid[position].is_empty? do
      @ui.show_error_message("Please choose a different position where the cell is empty\n")
      position = @ui.get_position
 end
```
Hmm. Maybe not break it all out, but it feels like there's a few different things going on that can be pulled out. The error clause, showing the error message, etc.

Reply:
thanks!